src/components/table-hoc/TableWithFilter.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import {keys} from 'ts-transformer-keys/index';
+import * as _ from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
+import {callIfDefined} from '../../utils/FalsyValuesUtils';
+import {ReduxConnect} from '../../utils/ReduxUtils';
+import {FilterBoxConnected} from '../filterBox/FilterBoxConnected';
+import {IFilterState} from '../filterBox/FilterBoxReducers';
+import {ITableHOCOwnProps} from './TableHOC';
+
+export interface ITableWithFilterConfig {
+    matchFilter?: (filterValue: string, datum: any) => boolean;
+}
+
+export interface ITableWithFilterStateProps {
+    filter: string;
+}
+
+export interface ITableWithFilterDispatchProps {
+    onRender: () => void;
+}
+
+export interface ITableWithFilterProps extends Partial<ITableWithFilterStateProps>,
+    Partial<ITableWithFilterDispatchProps>,
+    ITableHOCOwnProps {}
+
+const TableWithFilterPropsToOmit = keys<ITableWithFilterStateProps>();
+
+const defaultMatchFilter = (filter: string, datum: any) => JSON.stringify(_.values(datum).map((v: string) => v.toLowerCase())).indexOf(filter.toLowerCase()) !== -1;
+
+type FilterableTableComponent = React.ComponentClass<ITableWithFilterProps>;
+
+export const tableWithFilter = (config: ITableWithFilterConfig = {}) => (Component: FilterableTableComponent): FilterableTableComponent => {
+    const mapStateToProps = (state: IReactVaporState, ownProps: ITableWithFilterProps): ITableWithFilterStateProps | ITableHOCOwnProps => {
+        const filter: IFilterState = _.findWhere(state.filters, {id: ownProps.id});
+        const matchFilter = config.matchFilter || defaultMatchFilter;
+        const filterData = () => filter && filter.filterText
+            ? _.filter(ownProps.data, (datum: any) => matchFilter(filter.filterText, datum))
+            : ownProps.data;
+
+        return {
+            filter: filter && filter.filterText,
+            data: ownProps.data && filterData(),
+        };
+    };
+
+    @ReduxConnect(mapStateToProps)
+    class TableWithFilter extends React.Component<ITableWithFilterProps> {
+
+        componentDidUpdate(prevProps: ITableWithFilterProps) {
+            if (prevProps.filter !== this.props.filter) {
+                callIfDefined(this.props.onUpdate);

```

Comment:
just to be sure, we add this logic with the right props to compare on each component we want to add a behaviour to update on change ? eg. with a predicate component, we valide de props predicate value and call onUpdate if it change ?